src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java

```diff
@@ -33,4 +30,9 @@ public ResponseEntity<WalletDTO> createWallet(@Valid @RequestBody CreateWalletDT
 
         return new ResponseEntity<>(walletDTO, headers, HttpStatus.CREATED);
     }
+    @GetMapping(value = "/{id}")
+    public ResponseEntity<WalletDTO> findWalletById(@PathVariable long id) {
+        WalletDTO walletDTO = walletService.findById(id);
+        return new ResponseEntity<>(walletDTO, HttpStatus.OK);
```

Comment:
You can simplify this line to ``return new ResponseEntity<>(walletService.findById(id), HttpStatus.OK);``